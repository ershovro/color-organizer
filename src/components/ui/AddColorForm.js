import React from 'react';
import PropTypes from 'prop-types';
import '../../stylesheets/AddColorForm.scss';

const AddColorForm = ({ oneNewColor = f => f }) => {
   const onSubmitHandler = (event) => {
      event.preventDefault();
      oneNewColor(_textInput.value, _colorInput.value);
      _textInput.value = '';
      _colorInput.value = '#000000';
      _textInput.focus();
   }
   let _textInput, _colorInput;
   
   return (
      <form
         className="colorForm"
         onSubmit={onSubmitHandler}>
         
         <input
            ref={ textInput => _textInput = textInput }
            className="colorForm__textInput"
            type="text"
            placeholder="color title..."
            required
         />
         <input
            ref={ colorInput => _colorInput = colorInput }
            className="colorForm__colorInput"
            type="color"
         />
         <button className="colorForm__Button">
            Add
         </button>

      </form>
   )
}

AddColorForm.propTypes = {
   oneNewColor: PropTypes.func
}

export default AddColorForm