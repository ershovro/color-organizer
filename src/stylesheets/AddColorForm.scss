.colorForm {
   width: 98%;
   display: flex;
}

.colorForm__textInput {
   flex-grow: 1;
}

.colorForm__colorInput {
   padding: 0;
   height: +2em;
}

.colorForm__textInput, .colorForm__colorInput, .colorForm__Button {
    margin: .5em;
    padding: .25em;
    font-size: 1.2em;
    border-radius: 5px;
}

.colorForm__Button {
    background-color: green;
    color: white;
    border: none;
}