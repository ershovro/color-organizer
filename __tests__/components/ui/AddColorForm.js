import React from 'react';
import enzyme from 'enzyme';
import AddColorForm from "../../../src/components/ui/AddColorForm";
import { JSDOM } from 'jsdom';
import toJSON from 'enzyme-to-json'
import { compose } from 'redux';
import Adapter from 'enzyme-adapter-react-16';


const { window } = new JSDOM(`...`);
const { document } = (new JSDOM(`...`)).window;

enzyme.configure({ adapter: new Adapter() });
global.window = window;
global.document = document;


describe('<CAddColorForm /> ui component', () => {

   describe('Rendering ui', () => {

      it('Renders correctly', () => {
         compose(expect, toJSON, enzyme.shallow)( <AddColorForm /> ).
         toMatchSnapshot()
      });

      it('Renders a form with class .colorForm', () => {
         expect(
            enzyme.mount( <AddColorForm /> ).
            find('form.colorForm').
            length
         ).toBe(1);
      });

      it('Submiting default onNewColor does not cause error', () => {
         let _oneNewColor = jest.fn();
         let wrapper = enzyme.mount( <AddColorForm oneNewColor={_oneNewColor}/> );

         wrapper.find('input.colorForm__textInput').first().value = 'Best color';
         wrapper.find('input.colorForm__colorInput').first().value = "F0F0F0";
         wrapper.find('form.colorForm').simulate('submit');
      });

   });

   describe('Adding a new color', () => {
      let _oneNewColor = jest.fn();
      let wrapper;

      afterEach(() => {
         _oneNewColor.mockReset();
      });
      beforeEach(() => {
         wrapper = enzyme.mount( <AddColorForm oneNewColor={_oneNewColor}/> );
         wrapper.find('input.colorForm__textInput').instance().value = 'Best color';
         wrapper.find('input.colorForm__colorInput').instance().value = "#f0f0f0";
         wrapper.find('form.colorForm').simulate('submit');
      });

      it('Invoke onSubmit handler', () => {
         expect(_oneNewColor).toBeCalled();
      });

      it('Invoke onSubmit handler with correct arguments', () => {
         expect(_oneNewColor).toBeCalledWith('Best color', "#f0f0f0");
      });

      it('Reset the title value', () => {
         expect( wrapper.find('input.colorForm__textInput').instance().value ).
         toBe('')
      });

      it('Reset the color value', () => {
         expect(  wrapper.find('input.colorForm__colorInput').instance().value ).
         toBe('#000000')
      });
   });

});