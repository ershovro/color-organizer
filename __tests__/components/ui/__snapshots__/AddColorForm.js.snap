// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CAddColorForm /> ui component Rendering ui Renders correctly 1`] = `
<form
  className="colorForm"
  onSubmit={[Function]}
>
  <input
    className="colorForm__textInput"
    placeholder="color title..."
    required={true}
    type="text"
  />
  <input
    className="colorForm__colorInput"
    type="color"
  />
  <button
    className="colorForm__Button"
  >
    Add
  </button>
</form>
`;
